docs/advanced/psp/managing-plugins.md

```diff
@@ -0,0 +1,77 @@
+
+   Because the addresses of all associated contracts are now known, a static permission list can be emitted, hashed, and stored on-chain.
+
+4. Once the Plugin installation has been prepared, we use it as the parameter of the `applyInstallation()` action. Once encoded, this action is what must be added to the `Action[]` array of the installation proposal. That way, when the proposal passes, the action becomes executable and the plugin can be installed in the DAO using the parameters defined in the prepare installation process. For a plugin to be installed, it needs to be approved by the governance mechanism (plugin) of the organization, passed as the encoded action of a proposal, and executed by a signer.
+
+:::info
+The governance plugin can be a simple majority vote, an optimistic process or an admin governance plugin that does not involve a waiting period. It can be any governance mechanism existing within the DAO which has access to the DAO's `execute` permission.
+:::
+
+This gives the DAO members the opportunity to check which permissions the `PluginSetup` contract request before granting/revoking them.
+
+Plugin setup proposals must be carefully examined as they can be a potential security risk if the `PluginSetup` contract comes from an untrusted source. To learn more visit the [Security](./01-security-risk-mitigation.md) section.
+
+<!-- TODO: add a costs sections
+
+Optionally, the proposer can also request refunds for the gas spent for the preparation of the plugin in the proposal.
+-->
+
+#### What happens during the Preparation Application?
+
+After this initial preparation transaction, the addresses and permissions related to the plugin become apparent. The members of a governance plugin with permissions can decide if the installation proposal should be accepted or denied.
+
+Once the proposal has passed, the actions specified in the `Action[]` array can get executed and the `applyInstallation()` action is used to complete the installation of the plugin into the DAO.
+
+This is processed as follows:
+
+1. The DAO temporarily grants the `ROOT_PERMISSION_ID` permission to the `PluginSetupProcessor`. This is needed so that the processor can modify the DAO's permissions settings to set up the plugin.
+2. This `Action` calls the `applyInstallation` method in the `PluginSetupProcessor`, containing the list of requested permissions as argument. The permissions hash is compared with the stored hash to make sure that no permission was changed.
+   In addition to the above, the update process also upgrades the logic contract to which the proxy points too.
+3. If the hash is valid, the list is processed and `PluginSetupProcessor` conducts the requested sequence of `grant`, `grantWithCondition` and `revoke` calls on the owning DAO.
+   Finally, the `PluginSetupProcessor` asks the DAO to revoke the `ROOT_PERMISSION_ID` permission from itself.
```

Comment:
if I'm right this is misleading, the PSP doesn’t ask to remove the permission, but the Dao removes it. 

```suggestion
   4. Finally, the temporarily granted `ROOT_PERMISSION` to the `PluginSetupProcessor` is revoked.
```

Reply:
I'm seeing now that this is also in the documentation in production, 
<img width="833" alt="Screenshot 2024-06-07 at 12 46 05" src="https://github.com/aragon/developer-portal/assets/26546018/16714388-a757-4442-a27b-094d05cbeaec">

But, for installing the actions shouldn’t have a grant and a revoke ROOT permission for the PSP? 
I mean shouldn’t  the actions look like this: 
[ `grant(ROOT to PSP)`,
  `applyInstallation(...)`,
  `revoke(ROOT to PSP)`]
if that's the case the PSP is not asking to revoke his ROOT permission. 

I know this might be too specific and not needed for documentation purposes, but for me saying "_PSP ask to revoke ROOT from itself_" could be misunderstood, and thinking that the revoke is not needed on the actions since the PSP renounced to the permission

Reply:
@Rekard0 ^^